neo/IO/Helper.cs

```diff
@@ -129,11 +129,39 @@ public static T[] ReadSerializableArray<T>(this BinaryReader reader, int max = 0
             return array;
         }
 
+        public static uint[] ReadUIntArray(this BinaryReader reader, int max = 0x1000000)
+        {
+            uint[] array = new uint[reader.ReadVarInt((ulong) max)];
+            for (int i = 0; i < array.Length; i++) array[i] = reader.ReadUInt32();
+            return array;
+        }
+
         public static byte[] ReadVarBytes(this BinaryReader reader, int max = 0x1000000)
         {
             return reader.ReadBytes((int)reader.ReadVarInt((ulong)max));
         }
 
+        public static byte[][] ReadVarBytesArray(this BinaryReader reader, int maxItems=255, int maxItemLen = 1024)
+        {
+            byte items = (byte) reader.ReadVarInt((ulong)maxItems);
+            if (items > 0)
+            {
+                byte[][] output = new byte[items][];
+                for (int i = 0; i < items; i++)
+                {
+                    int bytes = (int) reader.ReadVarInt((ulong)maxItemLen);
+                    if (bytes == 0)
+                        output[i] = null;
+                    else
+                        output[i] = reader.ReadBytes(bytes);
+                }
+
+                return output;
+            }
+
+            return null;
```

Comment:
Maybe we should return a empty array?

Reply:
I need the behavior of having it be null if there are 0 items for the way I'm using it.

Reply:
I guess i could have it check for length 0 instead if I were to change it.

Reply:
However it comes in as null, and comes out as null. I think it is better the way it is.

Reply:
Feel free to re-open if you feel strongly about this, or message me on Discord to discuss.

Reply:
Maybe we can add a parameter `bool allowNull` to the method `ReadVarBytesArray()`.

Reply:
I like it. I'll add it.

Reply:
There could be some ambiguity though about whether to allow null for the VarBytes items in the array, or allow null for the entire result?
Originally I had named this something with `SupportingNull` at the end, because of its behavior.

Reply:
I'll call the parameter `bool shouldReturnNullIfEmpty`

Reply:
Done. I do think it is a little odd adding the feature when no code uses it yet. I generally try to avoid that, but since it is a generic helper that may have more usage in the future, I guess it is fine.